specs/models/Step.v1.json

```diff
@@ -0,0 +1,142 @@
+{
+  "title": "Step.v1",
+  "type": "object",
+  "properties": {
+    "processors": {
+      "type": "array",
+      "items": {
+        "type": "object",
+        "properties": {
+          "path": {
+            "type": "string",
+            "description": "Path to a module in the modules database that will be invoked via xdmp.invoke"
+          },
+          "when": {
+            "type": "string",
+            "description": "When the processor should be invoked. Only 'beforeContentPersisted' is supported."
+          },
+          "vars": {
+            "type": "object",
+            "description": "Any properties defined in this object are passed to the invoked module"
+          }
+        }
+      }
+    },
+    "customHook": {
+      "type": "object",
+      "properties": {
+        "module": {
+          "type": "string"
+        },
+        "parameters": {
+          "type": "object"
+        },
+        "user": {
+          "type": "string"
+        },
+        "runBefore": {
+          "type": "boolean"
+        }
+      }
+    },
+    "threadCount": {
+      "type": "number",
+      "description": "If set, overrides the threadCount defined at the flow level and in the step definition"
+    },
+    "batchSize": {
+      "type": "number",
+      "description": "If set, overrides the batchSize defined at the flow level and in the step definition"
+    },
+    "stepDefinitionType": {
+      "type": "string"
+    },
+    "stepDefinitionName": {
+      "type": "string"
+    },
+    "description": {
+      "type": "string",
+      "description": "Optional description fo the step"
+    },
+    "stepId": {
+      "type": "string",
+      "description": "This is generated on the server-side"
+    },
+    "headers": {
+      "type": "object",
+      "description": "Any properties in this object will be copied into the headers of each document processed by the step"
+    },
+    "validateEntity": {
+      "type": "boolean",
+      "description": "Applicable to mapping steps only"
+    },
+    "outputDatabase": {
+      "type": "string"
+    },
+    "inputDatabase": {
+      "type": "string"
+    },
+    "outputFormat": {
+      "type": "string",
+      "enum": [
+        "json",
+        "xml"
+      ]
+    },
+    "permissions": {
```

Comment:
Little tweak - I think "outputPermissions" here, for consistency with mlcp and to make it explicit how these will be applied.

Reply:
updated to use outputPermissions